addons/popochiu/engine/objects/character/popochiu_character.gd

```diff
@@ -108,6 +108,14 @@ var _looking_dir: int = Looking.DOWN
 var _animation_suffixes: Array = ['_d', '_l', '_r', '']
 # Holds the last PopochiuClickable that the character reached.
 var _last_reached_clickable: PopochiuClickable = null
+# Caches the last direction the character faced to avoid unnecessary recalculation
+var _last_direction_faced: int = -1
+# Caches the last animation played
+var _last_animation_requested: String = "animation_not_set"
+# Caches the last animation played
+var _last_animation_played: String = "animation_not_set"
```

Comment:
Reading the code below, I have the impression you can avoid using two variables for last requested and last played. The value of a single variable would be updated only if the user required a different animation than the last one played, right? I may be missing something obvious here...

Also, I would not use a magic string as "null value". It is VERY unlikely, but it can collide with an actual animation. `null` should do here, since a populated string, even if empty, should not be null.

Reply:
OK, makes sense. Sorry for not noticing it before!